src/decoder/mod.rs

```diff
@@ -1112,6 +1151,12 @@ impl<R: Read + Seek> Decoder<R> {
                     TiffUnsupportedError::UnsupportedBitsPerChannel(n),
                 )),
             },
+            SampleFormat::Int => match max_sample_bits {
+                n if n <=16 => DecodingResult::new_i16(buffer_size, &self.limits),
```

Comment:
The `n <= 8` case should be treated separately. Some code depends on the type of samples to evaluate the expected number of bytes. You could test that case by producing an `int8_rgb` image.